Mage.Sets/src/mage/cards/g/GollumSchemingGuide.java

```diff
@@ -0,0 +1,139 @@
+
+import mage.MageInt;
+import mage.abilities.Ability;
+import mage.abilities.common.AttacksTriggeredAbility;
+import mage.abilities.effects.OneShotEffect;
+import mage.abilities.effects.common.combat.CantBeBlockedSourceEffect;
+import mage.cards.*;
+import mage.constants.*;
+import mage.filter.FilterCard;
+import mage.filter.StaticFilters;
+import mage.game.Game;
+import mage.game.permanent.Permanent;
+import mage.players.Player;
+import mage.target.common.TargetOpponent;
+
+import java.util.UUID;
+
+/**
+ *
+ * @author Susucr
+ */
+public final class GollumSchemingGuide extends CardImpl {
+
+    public GollumSchemingGuide(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.CREATURE}, "{1}{B}");
+        
+        this.supertype.add(SuperType.LEGENDARY);
+        this.subtype.add(SubType.HALFLING);
+        this.subtype.add(SubType.HORROR);
+        this.power = new MageInt(2);
+        this.toughness = new MageInt(1);
+
+        // Whenever Gollum, Scheming Guide attacks, look at the top two cards of your library,
+        // put them back in any order, then choose land or nonland. An opponent guesses whether
+        // the top card of your library is the chosen kind. Reveal that card. If they guessed right,
+        // remove Gollum from combat. Otherwise, you draw a card and Gollum can't be blocked this turn.
+        this.addAbility(new AttacksTriggeredAbility(new GollumSchemingGuideEffect()));
+    }
+
+    private GollumSchemingGuide(final GollumSchemingGuide card) {
+        super(card);
+    }
+
+    @Override
+    public GollumSchemingGuide copy() {
+        return new GollumSchemingGuide(this);
+    }
+}
+
+class GollumSchemingGuideEffect extends OneShotEffect {
+
+    GollumSchemingGuideEffect(){
+        super(Outcome.Benefit);
+        this.staticText = "look at the top two cards of your library, " +
+            "put them back in any order, then choose land or nonland. " +
+            "An opponent guesses whether the top card of your library " +
+            "is the chosen kind. Reveal that card. If they guessed right, " +
+            "remove Gollum from combat. Otherwise, you draw a card and Gollum " +
+            "can't be blocked this turn";
+    }
+
+    private GollumSchemingGuideEffect(final GollumSchemingGuideEffect effect){
+        super(effect);
+    }
+
+    @Override public GollumSchemingGuideEffect copy(){
+        return new GollumSchemingGuideEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Player player = game.getPlayer(source.getControllerId());
+        if (player == null) {
+            return false;
+        }
+
+        Cards cards = new CardsImpl(player.getLibrary().getTopCards(game, 2));
+        // look at the top two cards of your library,
+        player.lookAtCards(source, null, cards, game);
+        // put them back in any order
+        PutCards.TOP_ANY.moveCards(player, cards, source, game);
```

Comment:
Should I disable temporarily the top deck card reveal effects when doing such reordering of top cards of the library? Similar to `LookLibraryControllerEffect`.

Reply:
Good question. It seems like a good idea to me, but only seems to be used in that one effect class.

Interestingly, the "you may look at the top card of your library any time" effects are handled by `ContinuousEffectImpl::canLookAtNextTopLibraryCard()` instead.

I recommend testing both with e.g. Sphinx of Jwar Isle and e.g. Oracle of Mul Daya, and seeing if info about the next card in library is shown when it shouldn't be. If so, then add the same logic as `LookLibraryControllerEffect` for now (might be better to refactor eventually so the engine handles automatically, if possible).

Reply:
Ok, there was no info leaked with the current code, with reveal top card effects (tried Future Sight & Sphinx of Jwar Isle), the revealed top does not update when the player is making moveCards choices.